index.js

```diff
@@ -157,18 +165,20 @@ class Pool extends EventEmitter {
         return result
       }
 
+      const queueCallback = (err, res, done) => {
+        clearTimeout(tid)
+        response.callback(err, res, done)
+      }
+
       // set connection timeout on checking out an existing client
       const tid = setTimeout(() => {
         // remove the callback from pending waiters because
         // we're going to call it with a timeout error
-        this._pendingQueue = this._pendingQueue.filter(cb => cb === response.callback)
+        remove(this._pendingQueue, queueCallback)
```

Comment:
Was `this._pendingQueue = this._pendingQueue.filter(cb => cb !== response.callback)` a valid fix too?